File: wear/src/main/java/at/shockbytes/corey/workout/PulseLogger.java
package at.shockbytes.corey.workout;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Martin Macheiner
 *         Date: 15.03.2017.
 */

public class PulseLogger {

    private List<Integer> list;

    public PulseLogger() {
        list = new ArrayList<>();
    }

    public void logPulse(int pulse) {
        if (pulse > 0) {
            list.add(pulse);
        }
    }

    public int getAveragePulse(boolean resetBuffer) {

        int sum = 0;

        if (list.size() == 0) {
            return 0;
        }

        for (Integer i : list) {
            sum += i;
        }
        int avg = sum / list.size();

        if (resetBuffer) {
            reset();
        }

        return avg;
    }

    public void reset() {
        list.clear();
    }

}

File: wear/src/main/java/at/shockbytes/corey/data/workout/PulseLogger.kt
package at.shockbytes.corey.data.workout

/**
 * Author:  Martin Macheiner
 * Date:    15.03.2017
 */
class PulseLogger {

    private val list: MutableList<Int> = mutableListOf()

    fun logPulse(pulse: Int) {
        if (pulse > 0) {
            list.add(pulse)
        }
    }

    fun getAveragePulse(resetBuffer: Boolean): Int {

        val avg = list.average().toInt()
        if (resetBuffer) {
            reset()
        }
        return avg
    }

    private fun reset() {
        list.clear()
    }
}
